sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -1073,6 +1078,71 @@ std::unique_ptr<CurlNetworkConnection> CurlConnectionPool::GetCurlConnection(Req
       60L * 60L * 24L,
       Details::c_DefaultFailedToGetNewConnectionTemplate + host);
 
+  /******************** Curl handle options apply to all connections created
+   * The keepAlive option is managed by the session directly.
+   */
+  if (options.proxy != "")
+  {
+    SetLibcurlOption(
+        newHandle,
+        CURLOPT_PROXY,
+        options.proxy.c_str(),
+        Details::c_DefaultFailedToGetNewConnectionTemplate + host
+            + ". Failed to set proxy to:" + options.proxy);
```

Comment:
Is this unnecessarily creating strings that won't be relevant in the happy path? So, now we are paying for the string creation all the time. Is such a pattern, expensive? Where are the strings allocated and how many?

Reply:
I believe these bits from the string becomes part of the application data.  About 25 bits here :)